prime-router/src/main/kotlin/fhirengine/utils/HL7Reader.kt

```diff
@@ -424,5 +424,42 @@ class HL7Reader(private val actionLogger: ActionLogger) : Logging {
                 else -> null
             }
         }
+
+        fun isAckMessage(message: Message): Boolean {
+            val acceptAcknowledgmentType = when (val structure = message[MSH_SEGMENT_NAME]) {
+                is NIST_MSH -> structure.msh15_AcceptAcknowledgmentType.value
+                is v27_MSH -> structure.msh15_AcceptAcknowledgmentType.value
+                is v251_MSH -> structure.msh15_AcceptAcknowledgmentType.value
+                else -> null
```

Comment:
Do we need to continue this pattern of listing out each message structure for each supported type? I'm not very familiar with how this code is actually working.

Reply:
I believe we do. My understanding is when the HL7 message is parsed, it's underlying classes are found within the specific version package. So we need to grab the right class to be able to dig in even if it happens to be the same structure.